src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
>the current checked-in implementation isn't validating the host name, so I've left it like that. Is that correct?

If this code is verifying the received SSL certificate from the server, then it is supposed to verify that the DNS name used for the request "matches" the name in the certificate.  And the "match" rules are complicated since it needs to verify X509 Subject field or the Subject Alternative Field as per the RFC.  If the name doesn't match, then `SslPolicyErrors.RemoteCertificateNameMismatch` is supposed to be added to the errors.